tests/functional/d/dataclass_with_default_factory.py

```diff
@@ -41,3 +42,15 @@ class Test2:
 
 
 Test2.int_prop["key"] = "value"  # [unsupported-assignment-operation]
+
+
+@dc.dataclass
+class TEST3:
+    """Test dataclass that puts call to field() in another function call"""
+    attribute: int = cast(int, field(default_factory=dict))
+
+
+@dc.dataclass
+class TEST4:
+    """Test dataclass that uses a non-annotated attribute"""
+    attribute = cast(int, field(default_factory=dict))
```

Comment:
You removed the test case from the astroid PR. Do you want to remove it here as well?